File: sdk/daena_sdk/exceptions.py
"""
Exception classes for Daena SDK.
"""


class DaenaAPIError(Exception):
    """Base exception for all Daena API errors."""
    
    def __init__(self, message: str, status_code: int = None, response: dict = None):
        super().__init__(message)
        self.message = message
        self.status_code = status_code
        self.response = response


class DaenaAuthenticationError(DaenaAPIError):
    """Raised when authentication fails."""
    pass


class DaenaRateLimitError(DaenaAPIError):
    """Raised when rate limit is exceeded."""
    
    def __init__(self, message: str, retry_after: int = None, **kwargs):
        super().__init__(message, **kwargs)
        self.retry_after = retry_after


class DaenaNotFoundError(DaenaAPIError):
    """Raised when a resource is not found."""
    pass


class DaenaValidationError(DaenaAPIError):
    """Raised when request validation fails."""
    pass


class DaenaTimeoutError(DaenaAPIError):
    """Raised when a request times out."""
    pass

File: sdk-js/src/exceptions.ts
/**
 * Exception classes for Daena SDK.
 */

export class DaenaAPIError extends Error {
  public readonly statusCode?: number;
  public readonly response?: any;

  constructor(message: string, statusCode?: number, response?: any) {
    super(message);
    this.name = 'DaenaAPIError';
    this.statusCode = statusCode;
    this.response = response;
    Object.setPrototypeOf(this, DaenaAPIError.prototype);
  }
}

export class DaenaAuthenticationError extends DaenaAPIError {
  constructor(message: string = 'Authentication failed. Please check your API key.', statusCode?: number, response?: any) {
    super(message, statusCode, response);
    this.name = 'DaenaAuthenticationError';
    Object.setPrototypeOf(this, DaenaAuthenticationError.prototype);
  }
}

export class DaenaRateLimitError extends DaenaAPIError {
  public readonly retryAfter?: number;

  constructor(message: string = 'Rate limit exceeded. Please try again later.', retryAfter?: number, statusCode?: number, response?: any) {
    super(message, statusCode, response);
    this.name = 'DaenaRateLimitError';
    this.retryAfter = retryAfter;
    Object.setPrototypeOf(this, DaenaRateLimitError.prototype);
  }
}

export class DaenaNotFoundError extends DaenaAPIError {
  constructor(message: string = 'Resource not found.', statusCode?: number, response?: any) {
    super(message, statusCode, response);
    this.name = 'DaenaNotFoundError';
    Object.setPrototypeOf(this, DaenaNotFoundError.prototype);
  }
}

export class DaenaValidationError extends DaenaAPIError {
  constructor(message: string = 'Request validation failed.', statusCode?: number, response?: any) {
    super(message, statusCode, response);
    this.name = 'DaenaValidationError';
    Object.setPrototypeOf(this, DaenaValidationError.prototype);
  }
}

export class DaenaTimeoutError extends DaenaAPIError {
  constructor(message: string = 'Request timed out.', statusCode?: number, response?: any) {
    super(message, statusCode, response);
    this.name = 'DaenaTimeoutError';
    Object.setPrototypeOf(this, DaenaTimeoutError.prototype);
  }
}

